pinot-spi/src/main/java/org/apache/pinot/spi/config/ConfigUtils.java

```diff
@@ -35,45 +36,55 @@ private ConfigUtils() {
   private static final Map<String, String> ENVIRONMENT_VARIABLES = System.getenv();
 
   /**
-   * Apply environment variables to any given BaseJsonConfig.
+   * Apply system properties and environment variables to any given BaseJsonConfig.
+   * Environment variables take precedence over system properties.
+   * Since the System properties are mutable, this method will read it at runtime.
    *
-   * @return Config with environment variable applied.
+   * @return Config with both system properties and environment variables applied.
    */
-  public static <T extends BaseJsonConfig> T applyConfigWithEnvVariables(T config) {
-    return applyConfigWithEnvVariables(ENVIRONMENT_VARIABLES, config);
+  public static <T extends BaseJsonConfig> T applyConfigWithEnvVariablesAndSystemProperties(T config) {
+    Map<String, String> combinedMap = new HashMap<>();
+    // Add all system properties to the map
+    System.getProperties().forEach((key, value) -> combinedMap.put(String.valueOf(key), String.valueOf(value)));
```

Comment:
Is the overhead high to dynamically load system properties for each config read?

Reply:
I'm thinking maybe we can periodically pull it, but we can iterate if we find this becomes blocker